### DISCUSSION TITLE
transact to FundMe.withdraw errored: Invalid parameters: must provide an Ethereum address.

### DISCUSSION THREAD

**Discussion by Krishduttm** [smartcontractkit/full-blockchain-solidity-course-py#918] [Q&A]
I'm getting this error when im trying to fund the contract from another account. Works fine with owner account but not the second one.

**Comment by n4n0b1t3** [comment]
It is difficult to reconstruct what might have gone wrong without context. Don't be shy to post your source code here, no paper to waste ;-)
Little tip: First click on the editors <> (code) symbol and then paste your code into the backticks. This helps with reading.

**Reply by Krishduttm** [reply]
Ive pasted the code, please check it out. Thanks

**Comment by Krishduttm** [comment]
```Solidity
// SPDX-License-Identifier: MIT

pragma solidity >=0.8.0;

import "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";

contract FundMe {

    mapping(address => uint256) public AddressToAmount;
    address[] public funders;
    //address of the owner (who deployed the contract)
    address public owner;

    constructor() {
        owner = msg.sender;
    }

    function Fund() public payable {
        uint256 minimumUSD = 50 * 10 ** 8;

        require(getConversionRate(msg.value) >= minimumUSD, "You need to spend more ETH!");

        AddressToAmount[msg.sender] += msg.value;
        funders.push(msg.sender);
    }


    function getPrice() public view returns(uint256) {
        AggregatorV3Interface priceFeed = AggregatorV3Interface(0x8A753747A1Fa494EC906cE90E9f37563A8AF630e);

        (,int256 answer,,,) = priceFeed.latestRoundData();

        return uint256(answer);
    }

    function getConversionRate (uint256 WeiAmount) public view returns(uint256) {
        
        uint256 EthPrice = getPrice(); 
        uint256 AmountInUsd = (WeiAmount * EthPrice) / 1000000000000000000;

        return AmountInUsd;

    }

    modifier onlyOwner {
    	//is the message sender owner of the contract?
        require(msg.sender == owner);
        _;
    }

    function withdraw () public onlyOwner payable {

        payable(msg.sender).transfer(address(this).balance);

        for (uint256 funderIndex=0; funderIndex < funders.length; funderIndex++){
            address funder = funders[funderIndex];
            AddressToAmount[funder] = 0;
        }
        //funders array will be initialized to 0
        funders = new address[](0);
    }


}
```

**Reply by n4n0b1t3** [reply]
... and the complete error message, please.

**Comment by n4n0b1t3** [comment]
OK, the source code is identical to https://github.com/PatrickAlphaC/fund_me/blob/main/FundMe.sol
Do you have MetaMask running and working? Do you have at least one account on one of the test networks with some test ETH in it?

**Reply by Krishduttm** [reply]
<img width="1440" alt="Screenshot 2022-02-02 at 7 05 41 PM" src="https://user-images.githubusercontent.com/59798012/152163978-bf34afad-4d12-4c91-a6b0-724f62d4373b.png">

I have two accounts in metamask, whichever account i use deploy with will fund successfully but if i switch to the other account to fund, it wont work.

**Reply by Krishduttm** [reply]
You can see in the console i deployed successfully but i got the error in the next line when i switched account and tried to fund

**Reply by cromewar** [reply]
Are you using injected web3 or Javascript virtual machine to deploy?

**Reply by Krishduttm** [reply]
im using injected web3

**Reply by cromewar** [reply]
Just as a test, delete the cache and cookies of your browser and try again.

**Reply by Krishduttm** [reply]
Tried this and still didn't work.
 NOTE -The above method will delete your remix files so make sure to backup the code locally

**Comment by n4n0b1t3** [comment]
Please check, if your second metamask account is connected. Here an example of a not connected account:
![image](https://user-images.githubusercontent.com/32838961/152173551-bdfd0893-a914-4f59-8d86-6b92c6fad8dc.png)

**Reply by Krishduttm** [reply]
It is connected :/

**Reply by brokenseal** [reply]
Exact same issue here

**Comment by cromewar** [comment]
As a workaround try removing the `onlyOwner` identifier for now, and give it a try,
